src/extension.ts

```diff
@@ -32,8 +34,24 @@ export async function activate(context: vscode.ExtensionContext): Promise<IExten
         promptForConnection: (ignoreFocusOut?: boolean) => {
             return controller.connectionManager.connectionUI.promptForConnection(ignoreFocusOut);
         },
-        listDatabases: (connectionInfo: IConnectionInfo) => {
-            return controller.connectionManager.listDatabases(connectionInfo);
+        connect: async (connectionInfo: IConnectionInfo) => {
+
+            const uri = utils.generateQueryUri().toString();
+            const connectionPromise = new Deferred<boolean>();
+            // First wait for initial connection request to succeed
+            const requestSucceeded = await controller.connectionManager.connect(uri, connectionInfo, connectionPromise);
+            if (!requestSucceeded) {
+                throw new Error(`Connection request for ${JSON.stringify(connectionInfo)} failed`);
```

Comment:
do these error message need to be localized?

Reply:
Generally we shouldn't be localizing error messages - instead if callers want to display the errors they should catch them and display a localized error message themselves.

Reply:
(but we're also not consistent about that at all so * shrug *. I'll look at the current connection functions and see what they do there)

Reply:
Yeah most other errors here aren't localized either so I'll keep with that being consistent. Generally the failure isn't going to be coming from these anyways - if the connection itself fails the promise we pass in to connect is going to reject and so that's the error that will be sent up.